addons/wowl/static/src/crash_manager/error_dialogs.ts

```diff
@@ -12,43 +12,86 @@ function capitalize(s: string | undefined): string {
 }
 
 interface ErrorDialogProps {
-  error: {
-    message?: string;
-    data?: {
-      debug?: string;
-    };
-    subType?: string;
-    traceback?: string;
-    type: "server" | "script";
-  };
+  name?: string;
+  message?: string;
+  traceback?: string;
 }
 
 export class ErrorDialog extends Component<ErrorDialogProps, OdooEnv> {
   static template = "wowl.ErrorDialog";
   static components = { Dialog };
-  title: string;
-  traceback: string;
+  title = this.env._t("Odoo Error");
+
+  state = useState({
+    showTraceback: false,
+  });
+
+  onClickClipboard() {
+    odoo.browser.navigator.clipboard.writeText(
+      `${this.props.name}\n${this.props.message}\n${this.props.traceback}`
+    );
+  }
+}
+
+export class ClientErrorDialog extends ErrorDialog {
+  title = this.env._t("Odoo Client Error");
+}
+
+export class ServerErrorDialog extends ErrorDialog {
+  title = this.env._t("Odoo Server Error");
+}
+
+export class NetworkErrorDialog extends ErrorDialog {
+  title = this.env._t("Odoo Network Error");
+}
+
+interface RPCErrorDialogProps {
+  name: string;
+  message?: string;
+  data?: {
+    [key: string]: any;
+  };
+  subType?: string;
+  traceback?: string;
+  type: "server" | "script" | "network";
+}
+
+export class RPCErrorDialog extends Component<RPCErrorDialogProps, OdooEnv> {
+  static template = "wowl.ErrorDialog";
+  static components = { Dialog };
+  title = this.env._t("Odoo Error");
+  traceback?: string;
+
   state = useState({
     showTraceback: false,
   });
 
   constructor() {
     super(...arguments);
-    const { data, message, subType, traceback, type } = this.props.error;
-    if (type === "server") {
-      this.title = capitalize(subType) || this.env._t("Odoo Error");
-    } else {
-      this.title = this.env._t("Odoo Client Error");
+
+    switch (this.props.type) {
```

Comment:
looks to me that this should be done by the code calling the error dialog

Reply:
I feel those lines would clutter the crash manager dispatcher for no gain if I moved it up.